packages/@aws-cdk/aws-certificatemanager/lib/dns_validated_certificate_handler/index.js

```diff
@@ -0,0 +1,196 @@
+    }).promise()
+  } while (describeCertResponse.Certificate.DomainValidationOptions[0].ValidationStatus !== 'PENDING_VALIDATION')
+
+  const record = describeCertResponse.Certificate.DomainValidationOptions[0].ResourceRecord
+
+  console.log(`Upserting DNS record into zone ${hostedZoneId}: ${record.Name} ${record.Type} ${record.Value}`)
+
+  const changeBatch = await route53.changeResourceRecordSets({
+    ChangeBatch: {
+      Changes: [{
+        Action: 'UPSERT',
+        ResourceRecordSet: {
+          Name: record.Name,
+          Type: record.Type,
+          TTL: 60,
+          ResourceRecords: [{
+            Value: record.Value
+          }]
+        }
+      }]
+    },
+    HostedZoneId: hostedZoneId
+  }).promise()
+
+  console.log('Waiting for DNS records to commit...')
+  await route53.waitFor('resourceRecordSetsChanged', {
+    // Wait up to 5 minutes
+    $waiter: {
+      delay: 30,
+      maxAttempts: 10
+    },
+    Id: changeBatch.ChangeInfo.Id
+  }).promise()
+
+  console.log('Waiting for validation...')
+  await acm.waitFor('certificateValidated', {
+    // Wait up to 5 minutes
+    $waiter: {
+      delay: 30,
+      maxAttempts: 10
+    },
+    CertificateArn: reqCertResponse.CertificateArn
+  }).promise()
+
+  return reqCertResponse.CertificateArn
+}
+
+/**
+ * Deletes a certificate from AWS Certificate Manager (ACM).
+ *
+ * @param {string} arn The certificate ARN
+ */
+const deleteCertificate = async function (arn) {
+  const acm = new aws.ACM()
+
+  console.log(`Deleting certificate ${arn}`)
+
+  await acm.deleteCertificate({
+    CertificateArn: arn
+  }).promise()
+}
+
+/**
+ * Main handler, invoked by Lambda
+ */
+exports.certificateRequestHandler = async function (event, context, callback) {
+  var responseData = {}
+  var physicalResourceId
+  var certificateArn
+
+  try {
+    switch (event.RequestType) {
+      case 'Create':
+        certificateArn = await requestCertificate(event.RequestId, event.ResourceProperties.DomainName, event.ResourceProperties.HostedZoneId)
+        responseData.Arn = physicalResourceId = certificateArn
+        break
+      case 'Delete':
+        physicalResourceId = event.PhysicalResourceId
+        // If the resource didn't create correctly, the physical resource ID won't be the
+        // certificate ARN, so don't try to delete it in that case.
+        if (physicalResourceId.startsWith('arn:')) {
+          await deleteCertificate(physicalResourceId)
+        }
+        break
+      case 'Update':
+        await deleteCertificate(event.PhysicalResourceId)
+        certificateArn = await requestCertificate(event.RequestId, event.ResourceProperties.DomainName, event.ResourceProperties.HostedZoneId)
+        responseData.Arn = physicalResourceId = certificateArn
+        break
+      default:
+        throw new Error(`Unsupported request type ${event.RequestType}`)
+    }
+
+    console.log(`Uploading SUCCESS response to S3...`)
+
+    await request({
```

Comment:
I believe this can be rewritten without an external dependency.

Reply:
It can, but using the requests library adds clarity and makes the code easier to read vs. using Node's built-in HTTP library.  I'd like to continue to use it if possible.

Reply:
Well here's the rub: we currently have no mechanism to do the 'npm install' at CDK build time, and I don't want to have people to run a manual step build anywhere in addition to requiring this library.

So either we get rid of the external dependencies and depend only on what comes with Node by default, or we cannot merge this upstream.

The source for a module that does (almost) everything we need to is here, and it doesn't have any additional dependencies: 

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-lambda-function-code.html#cfn-lambda-function-code-cfnresponsemodule

And the only thing it DOESN'T do is allow you to report a `Reason`  back to CloudFormation (silly, but that's the way it is). I'd prefer to copy/paste and change that file into here, until we have something better.